x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -320,6 +332,10 @@ export class EnterpriseSearchPlugin implements Plugin {
       visibleIn: [],
     });
 
+    share.url.locators.create<CreatIndexLocatorParams>(
+      new CreatIndexLocatorDefinition(searchPlayground.createIndexLocatorId)
```

Comment:
```suggestion
      new CreatIndexLocatorDefinition('CREATE_INDEX_LOCATOR_ID')
```

Referencing `searchPlayground` here inverts the dependency. the Locator is something `enterprise_search` defines and "owns" and is them consumed by `search_playground`.

I know it's not great to not have a shared constant between the usages of the locator, but we also don't want to add a direct dependency on `searchPlayground` for this string ID.

Reply:
I'm bit confused I thought ent_Search already dependent on search_playground according to this [file](https://github.com/elastic/kibana/pull/178049/files/3d99f87ef5e8f4adff8b5190537f5b459840ce11#diff-c05b8711f0f974638cce14eeed2c5efeaaf14566a5e5c4d6fc99ccd857b1e000)
So I thought as we have dependency on searchPlayground, and searchPlayground has locator dependency, it won't cost us anything, isn't it?

Reply:
The locator is for the create index url, we should not make that ID dependent on the search playground plugin IMO. The search playground wants the URL which is why we are defining the locator for it now. But is likely not to be the only plugin that could use the create index locator. and we don't want to require usage of this locator to being dependent on the search playground plugin.